src/components/SelectComponents/Autosuggest/Autosuggest.tsx

```diff
@@ -30,21 +30,32 @@ interface Props<S>
     inputPlaceholder: string;
     /** String to be shown when no suggestions are available */
     noSuggestionsPlaceholder?: string;
-    /** Defines if the first item of suggestions list is visible even while loading other elements */
-    isFirstItemAlwaysVisible?: boolean;
+    /** Defines if suggestions list is visible even while loading other elements */
+    allowMixingSuggestionsAndLoading?: boolean;
     /** Enable ListOptimizer component for decreasing render time */
     useOptimizeListRender?: boolean;
     /** Function to be called when a suggestion is removed  */
     onSelectionRemove?: (item: S) => void;
     /** Function to be called on submitting form */
     onSubmit?: () => void;
-    /** if suggestions are still loading, i.e. display placeholders */
+    /** If suggestions are still loading, i.e. display placeholders */
     isLoading?: boolean;
+    /** Means to overwrite the look and feel of the UI in its blurred state */
+    customBlurParams?: {
+        /** React node that will be rendered to indicate the current state of selections */
+        selectionIndicator: ReactNode;
+        /** Should the input field be shown. Normally should set this to true (hide) */
+        isInputHidden: boolean;
+        /** Should a clear button be shown on hover */
+        showClearButton: boolean;
```

Comment:
What do you think about making `showClearButton`, `isInputHidden `,  `selectionIndicator ` optional as well?

Reply:
I don't think we can make intelligent assumptions of what these would be. That's the whole point of it making a custom param.